db/db_iterator_test.cc

```diff
@@ -2544,6 +2544,124 @@ TEST_P(DBIteratorTest, RefreshWithSnapshot) {
   ASSERT_OK(db_->Close());
 }
 
+TEST_P(DBIteratorTest, AutoRefreshIterator) {
+  Options options = CurrentOptions();
+  options.disable_auto_compactions = true;
+  for (const bool auto_refresh_enabled : {false, true}) {
+    for (const bool use_snapshot : {false, true}) {
+      DestroyAndReopen(options);
+      // Multi dimensional iterator:
+      //
+      // L0 (level iterator): [key000000]
+      // L1 (table iterator): [key000001]
+      // Memtable           : [key000000, key000999]
+      for (int i = 0; i < 1002; i++) {
+        ASSERT_OK(Put(Key(i % 1000), "val" + std::to_string(i)));
+        if (i <= 1) {
+          ASSERT_OK(Flush());
+        }
+        if (i == 0) {
+          MoveFilesToLevel(1);
+        }
+      }
+
+      ReadOptions read_options;
+      read_options.snapshot = use_snapshot ? db_->GetSnapshot() : nullptr;
```

Comment:
I prefer to call this `explicit_snapshot` since the alternative still uses (implicit) snapshot